optics-core/src/Optics/ReadOnly.hs

```diff
@@ -8,32 +8,47 @@ import Optics.Internal.Profunctor
 
 -- | Class for read-write optics that have their read-only counterparts.
 class ToReadOnly k where
-  type ReadOnlyOptic k
   -- | Turn read-write optic into its read-only counterpart.
-  getting :: Optic k is s t a b -> Optic' (ReadOnlyOptic k) is s a
+  --
+  -- This is useful when you have an @optic :: Optic k is s t a b@ of read-write
+  -- kind @k@ such that @s@, @t@, @a@, @b@ are rigid, there is no evidence that
+  -- @s ~ t@ and @a ~ b@ and you want to pass @optic@ to one of the functions
+  -- that accept read-only optic kinds.
+  --
+  -- Example:
+  --
+  -- @
+  -- λ> let fstIntToChar = _1 :: Lens (Int, r) (Char, r) Int Char
+  -- λ> :t view fstIntToChar
+  --
+  -- <interactive>:1:6: error:
+  --     • Couldn't match type ‘Char’ with ‘Int’
+  --       Expected type: Optic' A_Lens NoIx (Int, r) Int
+  --         Actual type: Lens (Int, r) (Char, r) Int Char
+  --     • In the first argument of ‘view’, namely ‘fstIntToChar’
+  --       In the expression: view fstIntToChar
+  -- λ> :t view (getting fstIntToChar)
+  -- view (getting fstIntToChar) :: (Int, r) -> Int
```

Comment:
Thanks, this is a really helpful example.

Though it does make me wonder... the problem here could be solved by giving `view` a more general type, using `Optic` rather than `Optic'`. Is there a reason we shouldn't do that? It introduces two more type variables, of course, so perhaps they might sometimes be ambiguous?

Reply:
@adamgundry they will be ambiguous

Reply:
Actually I tried to make all of the read-only optic specific functions take and return `s t a b` instead of `s s a a`, though eventually I got ambiguous type variable errors in some auxiliary module (I think it was `Optics.Cons.Core`)